src/OpenLoco/Map/MapGenerator.cpp

```diff
@@ -427,14 +430,199 @@ namespace OpenLoco::Map::MapGenerator
+        call(0x0046A439, regs);
+        _heightMap = nullptr;
+    }
+
+    // 0x0046A5B3
+    static void generateTerrainOnMountains(HeightMap& heightMap, uint8_t surfaceStyle)
+    {
+        _heightMap = heightMap.data();
+        registers regs;
+        regs.ebx = surfaceStyle;
+        call(0x0046A5B3, regs);
+        _heightMap = nullptr;
+    }
+
+    // 0x0046A4F9
+    static void generateTerrainFarFromMountains(HeightMap& heightMap, uint8_t surfaceStyle)
+    {
+        _heightMap = heightMap.data();
+        registers regs;
+        regs.ebx = surfaceStyle;
+        call(0x0046A4F9, regs);
+        _heightMap = nullptr;
+    }
+
+    // 0x0046A0D8
+    static void generateTerrainInSmallRandomAreas(HeightMap& heightMap, uint8_t surfaceStyle)
     {
         _heightMap = heightMap.data();
-        call(0x0046A021);
+        registers regs;
+        regs.ebx = surfaceStyle;
+        call(0x0046A0D8, regs);
         _heightMap = nullptr;
     }
 
+    // 0x0046A227
+    static void generateTerrainInLargeRandomAreas(HeightMap& heightMap, uint8_t surfaceStyle)
+    {
+        _heightMap = heightMap.data();
+        registers regs;
+        regs.ebx = surfaceStyle;
+        call(0x0046A227, regs);
+        _heightMap = nullptr;
+    }
+
+    // 0x0046A66D
+    static void generateTerrainAroundCliffs(HeightMap& heightMap, uint8_t surfaceStyle)
+    {
+        _heightMap = heightMap.data();
+        registers regs;
+        regs.ebx = surfaceStyle;
+        call(0x0046A66D, regs);
+        _heightMap = nullptr;
+    }
+
+    static void generateTerrainNull(HeightMap& heightMap, uint8_t surfaceStyle) {}
+
+    using GenerateTerrainFunc = void (*)(HeightMap&, uint8_t);
+    static const GenerateTerrainFunc _generateFuncs[] = {
+        generateTerrainNull,
+        generateTerrainNull,
+        generateTerrainFarFromWater,
+        generateTerrainNearWater,
+        generateTerrainOnMountains,
+        generateTerrainFarFromMountains,
+        generateTerrainInSmallRandomAreas,
+        generateTerrainInLargeRandomAreas,
+        generateTerrainAroundCliffs
+    };
+
+    // 0x0046A021
+    static void generateTerrain(HeightMap& heightMap)
+    {
+        const auto style = getEverywhereSurfaceStyle();
+        if (!style.has_value())
+        {
+            return;
+        }
+
+        TilePosLoop tileLoop{ { 1, 1 }, { map_columns - 1, map_rows - 1 } };
+        for (auto& tilePos : tileLoop)
+        {
+            auto* surface = Map::TileManager::get(tilePos).surface();
+            if (surface == nullptr)
+            {
+                continue;
+            }
+            surface->setTerrain(*style);
+            surface->setVar6SLR5(0);
+
+            const auto variation = getRandomTerrainVariation(*surface);
+            if (variation.has_value())
+            {
+                surface->setVariation(*variation);
+            }
+        }
+
+        for (const auto pattern : {
+                 LandDistribuitionPattern::nowhere,
```

Comment:
is there any reason to include 'nowhere' here? it amounts to a no-op

Reply:
It was just too match vanilla. It can be removed

Reply:
fair enough, personally I don't see a reason to match vanilla here as it doesn't add anything, but of course I'll leave it up to you if you want to change it or not